gpytorch/kernels/kernel.py

```diff
@@ -261,6 +261,15 @@ def batch_shape(self) -> torch.Size:
     def batch_shape(self, val: torch.Size):
         self._batch_shape = val
 
+    @property
+    def device(self) -> Optional[torch.device]:
+        if self.has_lengthscale:
+            return self.lengthscale.device
+        else:
+            for param in self.parameters():
+                return param.device
```

Comment:
- What if `self.lengthscale` and parameters are on different devices? Can they be? Can other tensor attributes of a Kernel be on different devices?
- How about `self.active_dims` (when not None), since all Kernels should have that?

Reply:
Good point - this is modeled after the code in the `dtype` property below. I guess the standing assumption has been here that all the parameters (of which lenghtscale is one) are on the same device / have the same dtype. I can simplify this and raise an error if this assumption is violated.

Reply:
I retained the behavior if `has_lengthscale = True` for now to not cause any unforeseen BC issues. But I'm raising an error if parameters / devices are heterogeneous between the parameters.